src/EmbeddedFreeType/FreeTypeFontFileUpdator.class.st

```diff
@@ -0,0 +1,102 @@
+"
+update Free Types Fonts from a file.
+"
+Class {
+	#name : 'FreeTypeFontFileUpdator',
+	#superclass : 'AbstractFreeTypeFontFile',
+	#category : 'EmbeddedFreeType-Installation',
+	#package : 'EmbeddedFreeType',
+	#tag : 'Installation'
+}
+
+{ #category : 'instance creation' }
+FreeTypeFontFileUpdator class >> withFileInfoProvider: anFreeTypeFontProvider [
+
+	^ self new fileInfoProvider: anFreeTypeFontProvider 
+]
+
+{ #category : 'loading and updating' }
+FreeTypeFontFileUpdator >> addAndCacheFileInfoFromPath: path file: aFile [
+
+	| info |
+	info := self fileInfoFromPath: path file: aFile.
+	self fileInfoProvider addFileInfo: info index: index.
+	self fileInfoProvider cacheFileInfo: info index: index
+]
+
+{ #category : 'accessing' }
+FreeTypeFontFileUpdator >> cachedNumFacesFromData: bytes [
+
+	| cachedInfo cachedNumFaces |
+	[
+	(cachedInfo := self fileInfoProvider
+		               validCachedInfoFor: bytes
+		               index: index) isNotNil ] whileTrue: [
+		index = 0 ifTrue: [ cachedNumFaces := cachedInfo numFaces ].
+		self fileInfoProvider addFirstFileInfo: cachedInfo index: index.
+		index := index + 1 ].
+	^ cachedNumFaces
+]
+
+{ #category : 'error handling' }
+FreeTypeFontFileUpdator >> failedToOpen: face from: path index: i [
```

Comment:
The superclass has an instance variable called 'face'. Thus, this name cannot be used here and another name should be used.